tools/run_tests/performance/bq_upload_result.py

```diff
@@ -157,6 +173,42 @@ def _populate_metadata_inplace(scenario_result):
     scenario_result['metadata'] = metadata
 
 
+def _populate_metadata_from_file(scenario_result, test_metadata_file):
+    with open(test_metadata_file, 'r') as f:
+        test_metadata = json.loads(f.read())
+
+    # eliminate managedFields from metadata set
+    if 'managedFields' in test_metadata:
+        del test_metadata['managedFields']
+
+    # if use kubectl apply ..., kubectl will append current configuration to
+    # annotation, the field is deleted since it includes a lot of irrelevant
+    # information
+    annotations = test_metadata['annotations']
```

Comment:
I would not assume that test_metadata contains annotations, even if we expect that to be true. I would at least do `annotations = test_metadata.get('annotations', {})`

Reply:
Right, thank you!